tests/Unit/DataStructures/Tensor/EagerMath/Test_DotProduct.cpp

```diff
@@ -151,6 +151,10 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.EagerMath.DotProduct",
     CHECK_ITERABLE_APPROX(
         get(dot_product(three_d_covector_a, three_d_covector_b, inv_g)),
         DataVector(npts, 486.0));
+    const auto covector_dot_tensor = dot_product(three_d_covector_a, inv_g);
```

Comment:
more tests, more dimensions, more tensors (at least to rank 4) and some non-trivial symmetries... (e.g. Ijaa)